substrate-node/pallets/pallet-smart-contract/src/migration.rs

```diff
@@ -89,3 +184,233 @@ pub fn migrate_to_version_6<T: Config>() -> frame_support::weights::Weight {
         return 0;
     }
 }
+
+pub fn remove_deployment_contracts_from_contracts_to_bill<T: Config>() {
+    // we only bill capacity reservation contracts and name contracts
+    for index in 0..BillingFrequency::<T>::get() {
+        let mut contract_ids = ContractsToBillAt::<T>::get(index);
+        contract_ids.retain(|&id| {
+            let contract = Contracts::<T>::get(id).unwrap();
+            !matches!(contract.contract_type, types::ContractData::DeploymentContract(_))
+        });
+        ContractsToBillAt::<T>::insert(index, &contract_ids);
+    }
+}
+
+pub fn add_active_node_contracts<T: Config>() {
+    // active node contracts contains only capacity reservation contracts
+    ActiveNodeContracts::<T>::drain();
+    for (_contract_id, contract) in Contracts::<T>::iter() {
+        match contract.contract_type {
+            types::ContractData::CapacityReservationContract(crc) => {
+                let mut contracts = ActiveNodeContracts::<T>::get(crc.node_id);
+                contracts.push(contract.contract_id);
+                ActiveNodeContracts::<T>::insert(crc.node_id, &contracts);
+            }
+            _ => {}
+        }
+    }
+}
+
+pub fn add_used_resources<T: Config>() {
+    for (_contract_id, contract) in Contracts::<T>::iter() {
```

Comment:
I see the contracts already being iterated in the translation. Iterating more than once seems very unusefull and makes the migration slower than it has to be. I think it can be merged into where the translation happens.